jme3-core/src/main/java/com/jme3/environment/EnvironmentProbeControl.java

```diff
@@ -0,0 +1,127 @@
+package com.jme3.environment;
+
+import java.util.function.Function;
+
+import com.jme3.asset.AssetManager;
+import com.jme3.environment.baker.IBLGLEnvBakerLight;
+import com.jme3.environment.baker.IBLHybridEnvBakerLight;
+import com.jme3.light.LightProbe;
+import com.jme3.math.Vector3f;
+import com.jme3.renderer.RenderManager;
+import com.jme3.renderer.ViewPort;
+import com.jme3.scene.Geometry;
+import com.jme3.scene.Node;
+import com.jme3.scene.Spatial;
+import com.jme3.scene.control.Control;
+import com.jme3.texture.Image.Format;
+
+/**
+ * A control that automatically handles environment bake and rebake including
+ * only tagged spatials.
+ * 
+ * @author Riccardo Balbo
+ */
+public class EnvironmentProbeControl extends LightProbe implements Control {
+
+    private RenderManager renderManager;
+    private AssetManager assetManager;
+    private int envMapSize;
+    private Spatial spatial;
+    private boolean BAKE_NEEDED = true;
```

Comment:
When I see an `ALL_CAPS` name I expect that I'm looking at a constant. I find it slightly confusing that `BAKE_NEEDED` and `USE_GL_IR` are named this way but are variables.

Reply:
Fixed